hyperspy/misc/eels/tools.py

```diff
@@ -307,3 +308,61 @@ def eels_constant(s, zlp, t):
         data=(t * i0 / (332.5 * ke)) * np.log(1 + (beta * tgt / eaxis) ** 2))
     k.metadata.General.title = "EELS proportionality constant K"
     return k
+
+
+def get_edge_onset(data, start, end, x_axis, percent_position):
+    """Get the onset energy for a core loss edge.
+
+    Parameters
+    ----------
+    data : NumPy array
+    x_axis : NumPy array
+        Same size as data
+    start, end : scalar
+        Region where the edge onset will be found, in x_axis values.
+    percent_position : float
+        At what fraction of the ELNES signal the onset energy will be
+        found. See EELSModel.estimate_and_set_coreloss_edge_onset for more
+        information.
+
+    Returns
+    -------
+    onset_energy : scalar
+
+    Examples
+    --------
+    >>> from hyperspy.misc.eels.tools import get_edge_onset
+    >>> s = hs.datasets.artificial_data.get_core_loss_eels_signal()
+    >>> data, x_axis = s.data, s.axes_manager[0].axis
+    >>> onset = get_edge_onset(data, 600, 700, x_axis, 0.1)
+
+    """
+    if not ((percent_position >= 0.) and (percent_position <= 1.)):
+        raise ValueError(
+                "percent position ({0}) must be between 0 and 1".format(
+                    percent_position))
+    start_i = np.argmax(x_axis > start)
+    end_i = np.argmax(x_axis > end)
+    if end_i == 0:
+        end_i = len(x_axis)
+
+    data = data[start_i:end_i]
+    x_axis = x_axis[start_i:end_i]
+
+    data_max = data.max()
+    data_min = data.min()
+    data_max_i = data.argmax()
+    data_min_i = data.argmin()
```

Comment:
I would add here a sanity check: ``if data_max_i < data_min_i``. This may happen when analysing multi-dimensional datasets if the edge is not present at a given location. Actually, this sanity check could be used to assign a ``nan`` for the onset at those pixels.

Reply:
Good idea! However, I'm not sure how to add a `np.nan` to a `EELSCLEdge` component.
```python
Mn_L3.onset_energy.value = np.nan
```
Gives the error
```python
hyperspy/hyperspy/misc/eels/base_gos.py in get_qaxis_and_gos(self, ienergy, qmin, qmax)
     57         else:
     58             index = self.qaxis.searchsorted(qmax)
---> 59             g1, g2 = qgosi[index - 1:index + 1]
     60             q1, q2 = self.qaxis[index - 1: index + 1]
     61             gosqmax = get_linear_interpolation((q1, g1), (q2, g2), qmax)

ValueError: not enough values to unpack (expected 2, got 1)
```